util/src/main/java/io/grpc/util/AdvancedTlsX509KeyManager.java

```diff
@@ -39,20 +37,21 @@
 
 /**
  * AdvancedTlsX509KeyManager is an {@code X509ExtendedKeyManager} that allows users to configure
- * advanced TLS features, such as private key and certificate chain reloading, etc.
+ * advanced TLS features, such as private key and certificate chain reloading.
+ * We expect only one of {@link AdvancedTlsX509KeyManager#updateIdentityCredentials},
```

Comment:
Why was this restriction added? I know we talked about it some, but you can totally call some of these multiple times, and that's okay.

Reply:
Agreed, it's perfectly fine to call updateIdentityCredentials and updateIdentityCredentialsFromFile(File, File) (unless a user does it from multiple threads) multiple times.
I think that calling those 2 after updateIdentityCredentialsFromFile (File, File, long, TimeUnit, ScheduledExecutorService) is something we want to discourage - WDYT?

Reply:
As long as you call `close()` on the return value of `updateIdentityCredentialsFromFile(File, File, long, TimeUnit, ScheduledExecutorService)`, all the methods are reasonable again.

We can add to that one method "you must close() before calling other update methods"

Reply:
Makes sense, agreed. For example, if a user wants to

1. load initial creds from somewhere
2. keep updating creds from file
3. reload the creds urgently
4. continue updating creds from file
then it can be achieved by the following sequence

1. `updateIdentityCredentials`
2. `updateIdentityCredentialsFromFile(File, File, long, TimeUnit, ScheduledExecutorService)`
3. `close`
4. `updateIdentityCredentials`
5. `updateIdentityCredentialsFromFile(File, File, long, TimeUnit, ScheduledExecutorService)`

Updated the wording